From 3ace20102e9cd1a39cfb7375171a2eadff349a69 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:50:20 -0400
Subject: [tools] perf/symbols: Set freed members to NULL in dso destructor

Message-id: <1408953210-26343-133-git-send-email-jolsa@redhat.com>
Patchwork-id: 88858
O-Subject: [PATCH RHEL7.1 BZ1133083 132/322] perf symbols: Set freed members to NULL in dso destructor
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit ee021d42238daadc7ba49274bb0ba7dff219c6ab
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Tue Dec 10 15:26:55 2013 -0300

description
===========
To help in debugging use after free bugs.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/dso.c b/tools/perf/util/dso.c
index 2c7e189..19babb0 100644
--- a/tools/perf/util/dso.c
+++ b/tools/perf/util/dso.c
@@ -475,13 +475,23 @@ void dso__delete(struct dso *dso)
  int i;
  for (i = 0; i < MAP__NR_TYPES; ++i)
   symbols__delete(&dso->symbols[i]);
- if (dso->short_name_allocated)
+
+ if (dso->short_name_allocated) {
   free((char *)dso->short_name);
- if (dso->long_name_allocated)
+  dso->short_name    = NULL;
+  dso->short_name_allocated = false;
+ }
+
+ if (dso->long_name_allocated) {
   free((char *)dso->long_name);
+  dso->long_name   = NULL;
+  dso->long_name_allocated = false;
+ }
+
  dso_cache__free(&dso->cache);
  dso__free_a2l(dso);
  free(dso->symsrc_filename);
+ dso->symsrc_filename = NULL;
  free(dso);
 }
 
-- 
1.7.1